src/EditorFeatures/Core/Implementation/NavigationBar/NavigationBarController.cs

```diff
@@ -113,15 +121,24 @@ void connectToNewWorkspace()
                 {
                     await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
 
-                    connectToNewWorkspace();
+                    ConnectToNewWorkspace();
                 }).CompletesAsyncOperation(asyncToken);
             }
+
+            return;
+
+            void ConnectToNewWorkspace()
+            {
+                // For the first time you open the file, we'll start immediately
+                StartModelUpdateAndSelectedItemUpdateTasks(modelUpdateDelay: 0, selectedItemUpdateDelay: 0);
+            }
         }
 
         private void DisconnectFromWorkspace()
         {
             if (_workspace != null)
             {
+                _workspace.DocumentActiveContextChanged -= this.OnDocumentActiveContextChanged;
                 _workspace.WorkspaceChanged -= this.OnWorkspaceChanged;
```

Comment:
had to add this as WOrkspaceChanged doesn't tell us when the active doc changes.  Previously, we would kick off the work to handle that *when* a dropdown was expanded.  now that we're nto kicking of work on expansion, but only showing what info we have, this callback was needed.